LVT Cannot compile: Symbol has no implicit type

Hi Devs, 

I am trying to compile LVT (I have LDT and LIS compile just fine) and am using the mpicc and mpif90 compilers for this. I got past the huge module compilation for LDT by using a beefier machine and setting optimization level to 0. However, I am stuck at the following error, as I am not very familiar with fortran. 

`Compiling LVT

[INFO] Compiling LVT source code

/usr/bin/mpif90 -c -pass-exit-codes -ffree-line-length-0 -O0  -fbacktrace  -fconvert=big-endian -DHIDE_SHR_MSG -DNO_SHR_VMATH -DGFORTRAN -DHIDE_MPI -I/sandbox/esmf/install/mod  -I/usr/local/include/ -I/sandbox/netcdf-c/4.7.3/install/include/  -I/usr/local/include/  -I/usr/local/include/  -I./ -I../core/ ../datastreams/SMAPTB/readSMAPTBobs.F90
../datastreams/SMAPTB/readSMAPTBobs.F90:351:9:

  351 |      do c=1,LVT_rc%lnc
      |         1
Error: Symbol 'c' at (1) has no IMPLICIT type
../datastreams/SMAPTB/readSMAPTBobs.F90:350:6:

  350 |   do r=1,LVT_rc%lnr
      |      1
Error: Symbol 'r' at (1) has no IMPLICIT type
make: *** [Makefile:108: readSMAPTBobs.o] Error 1

[ERR] Compile failed
`
I've attached my configure.lvt file as well. As far as my understanding goes, this is an issue with the fortran code and maybe some of the right header files are missing, but it could also be an issue with my environment.

Any help is appreciated. 

[configure.lvt.txt](https://github.com/NASA-LIS/LISF/files/5152300/configure.lvt.txt)

--- a/lvt/datastreams/SMAPTB/readSMAPTBobs.F90
+++ b/lvt/datastreams/SMAPTB/readSMAPTBobs.F90
@@ -345,7 +345,6 @@ subroutine read_SMAPTB(source, filename, TBobs)
   deallocate(tbv_a_field)
   deallocate(tbh_f_field)
   deallocate(tbv_f_field)
-#endif
 
   do r=1,LVT_rc%lnr
      do c=1,LVT_rc%lnc
@@ -357,6 +356,7 @@ subroutine read_SMAPTB(source, filename, TBobs)
         endif
      enddo
   enddo
+#endif
 
 end subroutine read_SMAPTB
 

--- a/lvt/datastreams/SMAP_L3TB/readSMAP_L3TB.F90
+++ b/lvt/datastreams/SMAP_L3TB/readSMAP_L3TB.F90
@@ -191,6 +191,8 @@ subroutine read_SMAP_L3Tb(source, fname, L3TB)
 
   implicit none
 
+  integer                        :: source 
+  character(len=*)               :: fname
   real                           :: L3TB(LVT_rc%lnc,LVT_rc%lnr,4)
   
 !
@@ -209,8 +211,6 @@ subroutine read_SMAP_L3Tb(source, fname, L3TB)
 
 #if (defined USE_HDF5)
 
-  integer                       :: source 
-  character(len=*)              :: fname
   character*100,   parameter    :: Tb_gr_name = "Soil_Moisture_Retrieval_Data"
   character*100,   parameter    :: Tbv_field_name = "soil_moisture"
 

--- a/lvt/datastreams/SMAPvod/readSMAPvodobs.F90
+++ b/lvt/datastreams/SMAPvod/readSMAPvodobs.F90
@@ -398,7 +398,6 @@ subroutine read_SMAPvod(source, fname, vodobs)
 !BOP
 
 #if (defined USE_HDF5)
-
  
   character*100,   parameter    :: vod_gr_name = "Soil_Moisture_Retrieval_Data"
   character*100,   parameter    :: vod_field_name = "vegetation_opacity"

--- a/lvt/datastreams/SMAPvod/readSMAPvodobs.F90
+++ b/lvt/datastreams/SMAPvod/readSMAPvodobs.F90
@@ -379,6 +379,8 @@ subroutine read_SMAPvod(source, fname, vodobs)
 
   implicit none
 
+  integer                        :: source 
+  character(len=*)               :: fname
   real                           :: vodobs(LVT_rc%lnc,LVT_rc%lnr)
   
 !
@@ -397,8 +399,7 @@ subroutine read_SMAPvod(source, fname, vodobs)
 
 #if (defined USE_HDF5)
 
-  integer                       :: source 
-  character(len=*)              :: fname
+ 
   character*100,   parameter    :: vod_gr_name = "Soil_Moisture_Retrieval_Data"
   character*100,   parameter    :: vod_field_name = "vegetation_opacity"
 

